### ISSUE TITLE
TypeError: validAttr is not a function in NextJS SSR

### ISSUE THREAD

**Issue by alikazemkhanloo** [callstack/linaria#1095]
## Environment

versions:
```
    "@babel/core": "^7.19.6",
    "@linaria/babel-preset": "^4.2.1",
    "@linaria/core": "^4.2.1",
    "@linaria/react": "^4.2.1",
    "@linaria/webpack-loader": "^4.1.7",
    "babel-loader": "^8.2.5",
    "next": "12.3.1",
    "next-linaria": "^1.0.1-beta",
    "react": "18.2.0",
    "react-dom": "18.2.0"

```

- Linaria version: 4.2.1
- Bundler (+ version):  webpack 5 (default in nextjs)
- Node.js version: v16.13.0
- OS: Macos M1

## Description

I get this error when using linaria in nextjs (used `withLinaria` from `next-linaria`):
```
TypeError: validAttr is not a function

```
looks like an issue in `@linaria/react`'s styled. which uses `@emotion`'s is-prop-valid.

## Reproducible Demo

To reproduce the error
```js
//nextjs.config.js
/** @type {import('next').NextConfig} */
const withLinaria = require("next-linaria");

const nextConfig = {
  reactStrictMode: true,
  swcMinify: true,
};

module.exports = withLinaria(nextConfig);
```
```js
// babel.config.js
module.exports = {
  presets: ["next/babel", "@linaria/babel-preset"],
  plugins: [],
};
```
and just use styled from `@linaria/react` and refresh the page to see the error.

It is definitely an SSR issue. it is working in client side. ( I commented out the `styled` related code, saved the file so that the page would compile without error, and then added back the part of the code that I had commented out before. after saving the HMR updated the page and I could see that it was working just fine.)

related: #1094

**Comment by alikazemkhanloo** [comment]
As a workaround, I changed the code in [here](https://github.com/callstack/linaria/blob/master/packages/react/src/styled.ts#L69) from 

```js
 if (!validAttr(key)) {
```
to
```js
 // @ts-ignore
 const checkIfValid = validAttr?.default || validAttr;
 if (!checkIfValid(key)) {
```
and it worked without any problem.
I published the modified package as `linaria-react-fixed`. Still, I don't understand why this issue is happening. I would appreciate it if someone could shed some light on the cause of this error.
